index.js

```diff
@@ -83,9 +83,58 @@ class Jetpack {
       }
     };
 
-    this.hooks = {
+    // The `ServerlessEnterprisePlugin` awkwardly wreaks havoc with alternative
+    // packaging.
+    //
+    // `serverless` special cases it and ensures it's _always_ the last plugin.
+    // This means that our plugin order ends up looking like:
+    //
+    // - ...
+    // - 'Package',
+    // - ...
+    // - 'Jetpack',
+    // - 'ServerlessEnterprisePlugin'
+    //
+    // Unfortunately, the plugin hooks end up like this:
+    //
+    // - `Jetpack:before:package:createDeploymentArtifacts`:
+    //   Bundles all files, creating artifact to avoid `Package` doing it.
+    // - `ServerlessEnterprisePlugin:before:package:createDeploymentArtifacts`:
+    //   Creates the `s_<NAME>.js` wrapper files.
+    // - `Package:package:createDeploymentArtifacts`:
+    //   Creates any artifacts that don't already exist.
+    //
+    // This means that Jetpack can't easily get both the SFE wrappers and still
+    // control packaging:
+    //
+    // - If Jetpack stays with `before:package:createDeploymentArtifacts`
+    //   it misses the `s_<NAME>.js` wrapper files.
+    // - _But_, if Jetpack hooks to `package:createDeploymentArtifacts` then
+    //  `Package` will actually perform real packaging and it's too late.
+    //
+    // To get around this awkward situation, we create a hooks object that is
+    // delayed until the `initialize` lifecycle, then patched in last. This
+    // ensures that Jetpack's hooks run absolutely last for these events. This
+    // is still a bit hacky, but not nearly as invasive as some of the other
+    // approaches we considered. H/T to `medikoo` for the strategy:
+    // https://github.com/FormidableLabs/serverless-jetpack/pull/68#issuecomment-556987101
+    const delayedHooks = {
       "before:package:createDeploymentArtifacts": this.package.bind(this),
-      "before:package:function:package": this.package.bind(this),
+      "before:package:function:package": this.package.bind(this)
+    };
+
+    this.hooks = {
+      // Use delayed hooks to guarantee we are **last** to run so other things
+      // like the Serverless Enterprise plugin run before us.
+      initialize: () => {
+        const { hooks } = serverless.pluginManager;
+        Object.keys(delayedHooks).forEach((event) => {
+          hooks[event] = (hooks[event] || []).concat({
+            pluginName: this.constructor.name,
+            hook: delayedHooks[event]
+          });
+        });
+      },
```

Comment:
Looks good 👍

Reply:
Yay! Thanks for the review and suggestion!